--- a/net/minecraft/world/item/TridentItem.java
+++ b/net/minecraft/world/item/TridentItem.java
@@ -83,7 +_,7 @@
                         if (tridentSpinAttackStrength == 0.0F) {
-                                ThrownTrident::new, serverLevel, itemStack, player, 0.0F, 2.5F, 1.0F
+                                ThrownTrident::new, serverLevel, itemStack, player, 0.0F, 2.5F, (float) serverLevel.purpurConfig.tridentProjectileOffset  // Purpur - Projectile offset config
                             );
@@ -94,6 +_,7 @@
                                 return false;
                             }
+                            thrownTrident.setActualEnchantments(stack.getEnchantments()); // Purpur - Add an option to fix MC-3304 projectile looting
                             if (event.shouldConsume()) {
                             }
